iris-mpc-gpu/src/server/actor.rs

```diff
@@ -1586,8 +1583,23 @@ impl ServerActor {
         Ok((compact_device_queries, compact_device_sums))
     }
 
-    fn populate_or_policy_bitmap(&mut self, or_rule_serial_ids: Vec<Vec<u32>>, batch_size: usize) {
-        for (device_idx, _) in self.current_db_sizes.iter().enumerate() {
+    fn prepare_or_policy_bitmap(
+        &mut self,
+        or_rule_serial_ids: Vec<Vec<u32>>,
+        batch_size: usize,
+    ) -> Vec<CudaSlice<u64>> {
+        let streams = self
+            .device_manager
+            .devices()
+            .iter()
+            .map(|dev| dev.fork_default_stream().unwrap())
+            .collect::<Vec<_>>();
```

Comment:
no need to create new streams, you can just pass them into the function or use directly `self.streams[0]`. If you create new streams, it will just run async with the other stuff, which we don't want.